migrate-from-mariadb.md

```diff
@@ -0,0 +1,334 @@
+| mysql              | utf8mb3_general_ci |       67 |
+| mysql              | utf8mb4_bin        |        8 |
++--------------------+--------------------+----------+
+14 rows in set (0.045 sec)
+```
+
+See also [Character Set and Collation](/character-set-and-collation.md).
+
+## Dump and restore
+
+This method assumes that you take your application offline, migrate the data, and then re-configure your application to use the migrated data.
+
+It is strongly recommended to first do this on a test or development instance of your application before doing it in production. This is both to check for possible compatibility issues as to get insight into how much time the migration will take.
+
+1. Stop your application. Take your application offline. This ensures there are no modifications made to the data in MariaDB during or after the migration.
+
+2. Dump the data. For this the first step is to dump data in MariaDB with the `tiup dumpling` command.
+
+    ```shell
+    tiup dumpling --port 3306 --host 127.0.0.1 --user root --password secret -F 256MB  -o /data/backup
+    ```
+
+3. Restore the data. For this step we will use the `tiup tidb-lightning` command. See [Get Started with TiDB Lightning](/get-started-with-tidb-lightning.md) for how to configure TiDB Lightning and how to run it.
+
+4. Migrate user accounts and permissions. See [Export users and grants](#export-users-and-grants) for how to migrate your users and permissions.
+
+5. Reconfigure your application. You need to change the application configuration so that it can connect to the TiDB server.
+
+6. Clean up. Once you have verified that the migration is successful you can make a final backup of the data in MariaDB and stop the server. This also means you can stop and remove the DM cluster.
+
+## Replicate data
+
+This method assumes you would set up replication, stop your application and wait for the replication to catch up, and then re-configure your application to use TiDB.
+
+It is strongly recommended to first test your application before doing this migration in production.
+
+To use DM, you need to deploy a set of DM services either with a [TiUP cluster](/dm/deploy-a-dm-cluster-using-tiup.md) or with [TiDB Operator](/tidb-operator-overview.md). After that, use `dmctl` to configure the DM services.
+
+### Step 1. Prepare
+
+Make sure that binlogs are enabled on MariaDB and that the `binlog_format` is set to `ROW`. It is also recommended to set `binlog_annotate_row_events=OFF` and `log_bin_compress=OFF`.
+
+You also need an account with the `SUPER` permission or with the `BINLOG MONITOR` and `REPLICATION MASTER ADMIN` permissions. This account also needs read permission for the schemas you are going to migrate.
+
+If you are not using an account with the `SUPER` permission, then you might have to add the following to the DM configuration, because TiDB does not yet know how to check for MariaDB specific permissions.
+
+```yaml
+ignore-checking-items: ["replication_privilege"]
+```
```

Comment:
besides this MariaDB specific preparation, we can also mention DM's general precheck items.